Bridges/HttpKernel.php

```diff
@@ -63,23 +64,81 @@ public function bootstrap($appBootstrap, $appenv)
      */
     public function onRequest(ReactRequest $request, ReactResponse $response)
     {
-        if (null !== $this->application) {
-            try {
-                $syRequest = new SymfonyRequest();
-                $syRequest->headers->replace($request->getHeaders());
-                $syRequest->setMethod($request->getMethod());
-                $syRequest->server->set('REQUEST_URI', $request->getPath());
-                $syRequest->server->set('SERVER_NAME', explode(':', $request->getHeaders()['Host'])[0]);
-
-                $syResponse = $this->application->handle($syRequest);
-                $this->application->terminate($syRequest, $syResponse);
-
-                $headers = array_map('current', $syResponse->headers->all());
-                $response->writeHead($syResponse->getStatusCode(), $headers);
-                $response->end($syResponse->getContent());
-            } catch (\Exception $e) {
-                //
+        if (null === ($app = $this->application)) {
+            return;
+        }
+
+        $content = '';
+        $headers = $request->getHeaders();
+        $contentLength = isset($headers['Content-Length']) ? (int) $headers['Content-Length'] : 0;
+
+        $request->on('data', function($data)
+            use ($request, $response, &$content, $contentLength)
+        {
+            // read data (may be empty for GET request)
+            $content .= $data;
+
+            // handle request after receive
+            if (strlen($content) >= $contentLength) {
+                $syRequest = self::mapRequest($request, $content);
+
+                try {
+                    $syResponse = $app->handle($syRequest);
+                } catch (\Exception $exception) {
+                    $response->writeHead(500); // internal server error
+                    $response->end();
+                    return;
+                }
+                
+                self::mapResponse($response, $syResponse);
             }
+        });
+    }
+
+    /**
+     * Convert React\Http\Request to Symfony\Component\HttpFoundation\Request
+     * 
+     * @param ReactRequest $reactRequest
+     * @return SymfonyRequest $syRequest
+     */
+    protected static function mapRequest(ReactRequest $reactRequest, $content)
+    {
+        $method = $reactRequest->getMethod();
+        $headers = $reactRequest->getHeaders();
+        $query = $reactRequest->getQuery();
+        $post = array();
+
+        // parse body?
+        if (isset($headers['Content-Type']) && (0 === strpos($headers['Content-Type'], 'application/x-www-form-urlencoded')
+            && in_array(strtoupper($method), array('PUT', 'DELETE', 'PATCH'))
+        ) {
+            parse_str($content, $post);
```

Comment:
Thank you. Wondering though... why should we add multipart parsing in the Symfony bridge? Wouldn't it be task of symfony to take the request body and do whatever it needs with it?